### PR TITLE
Adds project suricata

### PR DESCRIPTION
**Pull request by catenacyber**
Asked in #624

The code is not yet merged in suricata main repository cf https://github.com/OISF/suricata/pull/3914

MSAN is not yet enabled because of Rust and C interactions cf https://github.com/google/oss-fuzz/issues/2145#issuecomment-485781098

@victorjulien could you ack ?

cc @kcc as you seemed interested in rust targets

### CODE REVIEW

**Comment by catenacyber** [comment]
Travis seems to fail because my corpus building bash magic is too verbose...

For instance, we have a corpus of about 50 000 rules and it gets computed with
`cat *.rules | while read l; do echo $l > corpus/$i.rule; i=$((i+1)); done`
https://github.com/google/oss-fuzz/pull/2533/files#diff-7507477856b797ccae1de19abe792c2bR64

What should we do about it ?

**Comment by inferno-chromium** [comment]
Can you redirect that output to >/dev/null. Log output is good to be readable. For future build breaks, it will make debugging harder with such big logs.

**Comment by catenacyber** [comment]
Just like that ?

**Comment by inferno-chromium** [comment]
> Just like that ?

You can verify locally with infra/helper.py.

**Comment by catenacyber** [comment]
> You can verify locally with infra/helper.py.

Yes, it is running...

**Comment by catenacyber** [comment]
This does not seem to work.
Actually, the command `cat *.rules | while read l; do echo $l > corpus/$i.rule; i=$((i+1)); done` does not print any output.
But it executes many subcommands that get logged :
```
+ echo alert tcp '$HOME_NET' any '->' '$EXTERNAL_NET' 23 '(msg:"ET' EXPLOIT Zollard PHP Exploit Telnet 'Outbound";' 'flow:to_server,established;' 'content:"/var/run/.zollard/";' 'reference:url,deependresearch.org/2013/12/hey-zollard-leave-my-internet-of-things.html;' 'classtype:attempted-user;' 'sid:2017800;' 'rev:2;' metadata:created_at 2013_12_04, updated_at '2013_12_04;)'
+ i=15343
+ read l
+ echo
+ i=15344
+ read l
+ echo alert tcp '$EXTERNAL_NET' any '->' '$HOME_NET' 23 '(msg:"ET' EXPLOIT Zollard PHP Exploit Telnet 'Inbound";' 'flow:to_server,established;' 'content:"/var/run/.zollard/";' 'reference:url,deependresearch.org/2013/12/hey-zollard-leave-my-internet-of-things.html;' 'classtype:attempted-user;' 'sid:2017799;' 'rev:2;' metadata:created_at 2013_12_04, updated_at '2013_12_04;)'
+ i=15345
+ read l
+ echo
+ i=15346
+ read l
```
So, I am trying something else

File: projects/suricata/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2018 Google Inc.
```
Comment (Dor1s) [inline projects/suricata/Dockerfile]: nit: 2019

Comment (Dor1s) [reply projects/suricata/Dockerfile]: in the other files as well :)

File: projects/suricata/build.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+cd suricata
+sh autogen.sh
+./configure --disable-shared --enable-fuzztargets
+make
+
+cp src/fuzz_* $OUT/
+
+# build corpuses
+# default configuration file
+zip -r $OUT/fuzz_confyamlloadstring_seed_corpus.zip suricata.yaml
```
Comment (Dor1s) [inline projects/suricata/build.sh]: is `suricata.yaml` a directory? If not, are there any other potential seed inputs?

**Comment by catenacyber** [comment]
> I don't see @victorjulien on the upstream contributors page: https://github.com/OISF/suricata/graphs/contributors

He is the main contributor but as his profile says https://github.com/inliniac moved to https://github.com/victorjulien

File: projects/suricata/Dockerfile (hunk 1 above)
Comment (catenacyber) [reply projects/suricata/Dockerfile]: ok

File: projects/suricata/build.sh (hunk 2 above)
Comment (catenacyber) [reply projects/suricata/build.sh]: > is suricata.yaml a directory? 

No this is a single file (the default configuration)
I do not think of other available inputs...

**Comment by victorjulien** [comment]
Please hold on this. I would like to make sure we have the necessary support merged in our git repo first. Also, due to team availability over the summer, it would work a lot better for us to start this in the fall.

**Comment by catenacyber** [comment]
@victorjulien could you approve this now that we merged the fuzz targets in upstream Suricata repository ? Thanks :-)

**Comment by catenacyber** [comment]
Still needs https://github.com/OISF/suricata/pull/4732 before merging

**Comment by victorjulien** [comment]
I have just merged https://github.com/OISF/suricata/pull/4732, so upstream is ready.

Approved from the Suricata project.

**Comment by catenacyber** [comment]
Thanks @Dor1s 

The language is actually C and Rust
How does this new attribute matter ?

**Comment by Dor1s** [comment]
C or C++ doesn't matter, we use C++ by default as libFuzzer is C++ and is linked in :) I think fuzz targets are written in C, so `Rust` value is not really applicable here?

**Comment by Dor1s** [comment]
> How does this new attribute matter ?

As of now it just skips coverage job for non C/C++ projects, but in future it will guide other pieces of OSS-Fuzz such as bad build checks, srcmap generation, and others. See https://github.com/google/oss-fuzz/issues/3297 and https://github.com/google/oss-fuzz/pull/3477 if you're curious to learn more :)

**Comment by catenacyber** [comment]
> I think fuzz targets are written in C, so Rust value is not really applicable here?

I do not know. The C fuzz targets call Rust code at some points

> As of now it just skips coverage job for non C/C++ projects

Hope you have time to review #3142 soon ;-)

**Pull request merged**